### ISSUE TITLE
Wondering why many sharp peaks created in the deeptools 2.0 heatmap profile

### ISSUE THREAD

**Issue by zx0223winner** [deeptools/deepTools#1348]
**Welcome to deepTools GitHub repository! Before opening the issue please check
that the following requirements are met :**

 - [ ] Search whether this issue (or a similar issue) has been solved before
 using the search tab above. Link the previous issue if appropriate below.

 - [ ] Paste your deepTools version (`deeptools --version`) and your python
 version (`python --version`) below.

 - [ ] Paste the full deepTools command that produces the issue below
 (ignore if you simply spotted the issue in the code/documentation).

 - [ ] Paste the output printed on screen from the command that produces the issue
 below (ignore if you simply spotted the issue in the code/documentation).

![sample](https://github.com/user-attachments/assets/0334ce59-2b7d-440f-aa3f-5a4d30866160)

computeMatrix reference-point \
	--referencePoint TSS \
	-b 3000 \
	-a 3000 \
	-R {input.bed} \
	-S {input.bigwig} \
	--binSize 50 \
	-o {output} \
	--outFileSortedRegions {params.dir1} \

plotHeatmap \
	-m {input} \
	-out {output} \
	--interpolationMethod auto \
	--plotTitle {params} \
	--colorList 'yellow, red' 'green, blue' \

plotProfile \
	-m {input} \
	-out {output} \
	--plotTitle {params} \

**Comment by zx0223winner** [comment]
@WardDeb

**Comment by ckuenne** [comment]
just guessing, but maybe you have regions with extreme outlier bins that drive the average up. try to just plot the top50 regions.

**Comment by sunta3iouxos** [comment]
could those be mononucleosomes?
there is the one before the TSS and the 3 after that.
What kind of data are those?

**Comment by miaoxiannv** [comment]
Hi there, I was wondering if you managed to solve this issue? I’m encountering the same problem. If you found a solution, would you mind sharing how you fixed it? Any advice would be appreciated. Thanks!

**Comment by ckuenne** [comment]
hi, it's almost certainly a couple of extreme genes driving this. try to remove them from the matrix before plotting:


```
computeMatrixOperations dataRange ...
computeMatrixOperations filterValues ... --max XY
```

**Comment by zx0223winner** [comment]
@ckuenne 
> just guessing, but maybe you have regions with extreme outlier bins that drive the average up. try to just plot the top50 regions.

yes, there are some extreme outliers, one way to fix is rank the gene region score from ComputeMatrix file which calculates scores per genome regions. Get rid of those gene with extremely low regions.

> hi, it's almost certainly a couple of extreme genes driving this. try to remove them from the matrix before plotting:
computeMatrixOperations dataRange ...
computeMatrixOperations filterValues ... --max XY

@miaoxiannv 
I have not tried the above "computeMatrixOperations" commandline, but the idea is same trying to filter out those outliers. You can also normalize the gene region, for example, in your matrix, you set bin size 50bp, then 120 bins across 6kb region. Set a mean threshold to remove those extreme genes with very small normalized region values (bin scores). Normalizing each bin value by dividing the mean. Recreate heatmap profile with filtered computeMatrix should fix it.
